src/diplomas/tests/tests_diploma_regenerator.py

```diff
@@ -1,71 +1,103 @@
 import pytest
 
 from diplomas import tasks
-from diplomas.models import DiplomaTemplate
+from diplomas.models import Diploma
 from diplomas.services import DiplomaRegenerator
 
 pytestmark = [
     pytest.mark.django_db,
-    pytest.mark.skip(reason='fixme'),
 ]
 
 
 @pytest.fixture(autouse=True)
-def diploma_generator(mocker):
-    return mocker.patch('diplomas.services.diploma_regenerator.DiplomaGenerator.__init__', return_value=None)
+def _mock_diploma_generator_fetch_image(mocker, factory):
+    image = factory.uploaded_image()
+    mocker.patch('diplomas.services.diploma_regenerator.DiplomaGenerator.fetch_image', return_value=image)
 
 
 @pytest.fixture(autouse=True)
-def run_diploma_generation(mocker):
-    return mocker.patch('diplomas.services.diploma_generator.DiplomaGenerator.__call__')
+def template_en(mixer, course):
+    return mixer.blend('diplomas.DiplomaTemplate', slug='test-template', course=course, language='EN', homework_accepted=False)
 
 
-@pytest.fixture(autouse=True)
+@pytest.fixture
 def diploma_ru(mixer, order):
-    return mixer.blend('diplomas.Diploma', study=order.study, language='ru')
+    return mixer.blend('diplomas.Diploma', study=order.study, language='RU')
 
 
-@pytest.fixture(autouse=True)
+@pytest.fixture
 def diploma_en(mixer, order):
-    return mixer.blend('diplomas.Diploma', study=order.study, language='en')
+    return mixer.blend('diplomas.Diploma', study=order.study, language='EN')
 
 
 @pytest.fixture
-def send_mail(mocker):
-    return mocker.patch('mailing.tasks.Owl.send')
+def mock_diploma_generator(mocker):
+    return mocker.patch('diplomas.services.diploma_generator.DiplomaGenerator.__call__', autospec=True)
 
 
-def test_diplomas_are_regenerated(student, course, diploma_generator, mocker, order):
+@pytest.fixture
+def mock_diploma_regenerator(mocker):
+    return mocker.patch('diplomas.services.diploma_regenerator.DiplomaRegenerator.__call__', autospec=True)
+
+
+def test_diplomas_are_regenerated(student, course, diploma_ru, diploma_en, order):
     DiplomaRegenerator(student)()
 
-    diploma_generator.assert_has_calls(mocker.call(course=course, student=student, language='ru'))
-    diploma_generator.assert_has_calls(mocker.call(course=course, student=student, language='en'))
+    diploma_ru.refresh_from_db()
+    assert diploma_ru.modified is not None
+    diploma_en.refresh_from_db()
+    assert diploma_en.modified is not None
+
 
+def test_task_diplomas_regenerated(student, course, order, diploma_ru, diploma_en):
+    tasks.regenerate_diplomas.delay(student_id=student.id)
+
+    diploma_ru.refresh_from_db()
```

Comment:
```suggestion
    diploma_ru.refresh_from_db()
    diploma_en.refresh_from_db()
    
    assert diploma_ru.modified is not None
    assert diploma_en.modified is not None
```

Кажется так тесты будет чуть легче считать.

Reply:
Объединил `refresh_from_db()` и `assert` друг по другом.

Что делать не стал: разделять `refresh_from_db()` от `assert` пробелом. Если так сделать, то блок с `refresh` будет выглядеть как **act** хотя это уже **assert**.

В общем сейчас не стал разделять, но если считаешь, что так будет лучше и не будет вводить в заблуждение — поддержку и добавлю.